php-checks/src/main/java/org/sonar/php/checks/security/RegexUsageCheck.java

```diff
@@ -63,9 +74,37 @@
 
   @Override
   protected void createIssue(FunctionCallTree tree) {
-    if (!tree.arguments().isEmpty()) {
-      context().newIssue(this, tree, MESSAGE);
+    int index = getPatternArgumentIndex(tree);
+
+    if (tree.arguments().size() <= index) {
+      return;
+    }
+
+    ExpressionTree patternArgument = tree.arguments().get(index);
+    if (patternArgument.is(Kind.REGULAR_STRING_LITERAL)) {
+      String value = ((LiteralTree) patternArgument).value();
+      if (value.length() >= MIN_PATTERN_LENGTH && hasEnoughNumberOfSpecialChars(value)) {
+        context().newIssue(this, tree, MESSAGE);
+      }
+    }
+  }
+
+  private int getPatternArgumentIndex(FunctionCallTree tree) {
+    int index = 0;
+    if (tree.callee().toString().equalsIgnoreCase(MB_EREG_SEARCH_INIT)) {
+      index = 1;
+    }
+    return index;
```

Comment:
personal preference (so up to you to take the suggestion into account) but I would go for : 
```
if(cond) {
 return 1;
}
return 0;